pkg/link/opstate.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2024 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package link
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/vishvananda/netlink"
+)
+
+func WaitForOperStateUp(linkName string) (netlink.Link, error) {
+	var link netlink.Link
+	var err error
+	retries := []int{0, 50, 500, 1000, 1000}
+	for idx, sleep := range retries {
+		time.Sleep(time.Duration(sleep) * time.Millisecond)
+
+		link, err = netlink.LinkByName(linkName)
+		if err != nil {
+			return nil, err
+		}
+		if link.Attrs().OperState == netlink.OperUp {
+			break
+		}
+
+		if idx == len(retries)-1 {
+			return nil, fmt.Errorf("timeout waiting for %q state up", linkName)
+		}
+	}
+	return link, nil
+}
+
+func SetUp(linkName string) error {
+	link, err := netlink.LinkByName(linkName)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve link: %v", err)
```

Comment:
nit: please use `%w` to wrap errors everywhere.

Reply:
I didn't saw error being unwrapped in the bridge plugin code, also we didnt warped them before this change :slightly_smiling_face: 
Why is it necessary to wrap them here?

Reply:
because all of this code was written before `%w` was a thing :-).

Reply:
Done